third_party/move/move-bytecode-verifier/src/acquires_list_verifier.rs

```diff
@@ -202,6 +210,29 @@ impl<'a> AcquiresVerifier<'a> {
         Ok(())
     }
 
+    fn ld_function_acquire(
+        &mut self,
+        fh_idx: FunctionHandleIndex,
+        offset: CodeOffset,
+    ) -> PartialVMResult<()> {
+        // Currenty we are disallowing acquires for any function value which
+        // is created, so InvokeFunction does nothing with acquires.
+        // TODO(LAMBDA) In the future this may change.
+        let function_handle = self.module.function_handle_at(fh_idx);
+        let function_acquired_resources = self.function_acquired_resources(function_handle, fh_idx);
+        if !function_acquired_resources.is_empty() {
```

Comment:
We disallow both reentrancy and nonempty acquires lists for now, so it's definitely safe, if a bit limited.  Your foreign move module can't call back into this one, since you're already on the stack.

In the future we can add more general access_specifiers to functions and function types, and check that the dynamic call doesn't exceed the accesses of the caller.  I'd rather work that out first, but people are anxious about an MVP deadline.

Reply:
I was being overly cautious here.   You're right, as long as we disallow all dynamic reentrancy we can ignore the acquires list of the function here.

Let's discuss post-MVP approaches to loosen constraints later.